code/Controllers/CMSMain.php

```diff
@@ -492,7 +492,7 @@ public function TreeAsUL()
         $modelClass = $this->getModelClass();
 
         DataObject::singleton($modelClass)->prepopulateTreeDataCache(null, [
-            'childrenMethod' => 'AllChildrenIncludingDeleted',
+            'childrenMethod' => Hierarchy::config()->get('default_children_method'),
```

Comment:
Given all the work we did to generalise `CMSMain` to pave the way for using it as a base for a general tree admin for any `Hierarchical` model, I think we should allow the base model to set the configuration.

This allows using different methods for different hierarchical models.
It also means we don't have to have `Configurable` on the extension which is a bit of a weird pattern.

`$baseClass = DataObject::getSchema()->baseDataClass($modelClass);`

```suggestion
            'childrenMethod' => $baseClass::config()->get('default_children_method'),
```

Reply:
Please also add tests that validate this respects config from the correct place.

Reply:
I think this would just be `$modelClass::config()->get('default_children_method')` - because a subclass may override the config on base class?

I think we'd still want to have `private static $default_children_method` defined on Hierarchy otherwise we have config which isn't defined anywhere, which is a pretty confusing pattern

Reply:
Agree define the config there. Just doesn't need to use the `Configurable` trait.

> I think this would just be `$modelClass::config()->get('default_children_method')` - because a subclass may override the config on base class?

Right, but the queries aren't checking each class, they're querying based on the base class. So the base class is the only one that really matters. The PHPDoc for the config should make it clear that only the base class values will be respected, and then if someone sets it on the base class they're just not following what is documented and should expect failure.

Reply:
OK base class makes sense

Though we'll still need a keep the config on Hierarchy as there needs to be a default defined somewhere because we cannot assume that all classes that apply the Hierarchy extension will have the config defined

Reply:
Again:

> Agree define the config there. Just doesn't need to use the `Configurable` trait.

You don't need that trait on an extension to provide default configuration